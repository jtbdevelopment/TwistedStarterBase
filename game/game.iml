<?xml version="1.0" encoding="UTF-8"?>
<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
  <component name="FacetManager">
    <facet type="Spring" name="Spring">
      <configuration />
    </facet>
  </component>
  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_5">
    <output url="file://$MODULE_DIR$/target/classes" />
    <output-test url="file://$MODULE_DIR$/target/test-classes" />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/src/main/groovy" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/src/main/resources" type="java-resource" />
      <sourceFolder url="file://$MODULE_DIR$/src/test/groovy" isTestSource="true" />
      <excludeFolder url="file://$MODULE_DIR$/target" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-mongo:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-hazelcast:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-events:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web-std-multiplayer:1.3" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-websocket:1.3" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-web:test-jar:tests:1.3" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-mongo:test-jar:tests:1.3" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-dev-utilities:1.3" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:javax-websocket-server-impl:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-annotations:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-plus:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-jndi:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.ow2.asm:asm:5.0.1" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.ow2.asm:asm-commons:5.0.1" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.ow2.asm:asm-tree:5.0.1" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:javax-websocket-client-impl:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: javax.websocket:javax.websocket-api:1.0" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-mongo:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-hazelcast:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-events:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web-std-multiplayer:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring-web-mongo:1.7" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring-mongo:1.7" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-websocket:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-std-tracker:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-std-maintenance:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-web:test-jar:tests:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-mongo:test-jar:tests:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-dev-utilities:1.4-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-mongo:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-hazelcast:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-web-std-multiplayer:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring-web-mongo:1.9" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring-web:1.9" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring:1.9" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-tx:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework.data:spring-data-commons:1.13.0.RELEASE" level="project" />
    <orderEntry type="library" scope="RUNTIME" name="Maven: org.slf4j:jcl-over-slf4j:1.7.22" level="project" />
    <orderEntry type="library" name="Maven: org.springframework.security:spring-security-web:4.2.1.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: aopalliance:aopalliance:1.0" level="project" />
    <orderEntry type="library" name="Maven: org.springframework.security:spring-security-core:4.2.1.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-web:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework.social:spring-social-core:1.1.4.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-common:core-spring-mongo:1.9" level="project" />
    <orderEntry type="library" name="Maven: org.springframework.data:spring-data-mongodb:1.10.0.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.mongodb:mongo-java-driver:2.14.3" level="project" />
    <orderEntry type="library" name="Maven: org.codehaus.gpars:gpars:1.2.1" level="project" />
    <orderEntry type="library" name="Maven: org.multiverse:multiverse-core:0.7.0" level="project" />
    <orderEntry type="library" name="Maven: org.codehaus.jsr166-mirror:jsr166y:1.7.0" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-core:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: commons-logging:commons-logging:1.2" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-context:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-aop:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-beans:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.springframework:spring-expression:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.slf4j:slf4j-api:1.7.22" level="project" />
    <orderEntry type="library" name="Maven: ch.qos.logback:logback-classic:1.1.9" level="project" />
    <orderEntry type="library" name="Maven: ch.qos.logback:logback-core:1.1.9" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-websocket:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-std-tracker:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" name="Maven: com.jtbdevelopment.core-games:games-std-maintenance:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-web:test-jar:tests:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-mongo:test-jar:tests:1.8-SNAPSHOT" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games-dev-utilities:1.7" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-games:games:1.7" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: commons-codec:commons-codec:1.10" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: com.jtbdevelopment.core-common:core-spring-mongo:test-jar:tests:1.9" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-servlet:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-security:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-server:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-webapp:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-xml:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-util:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:websocket-server:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:websocket-common:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:websocket-api:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-io:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:websocket-client:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty.websocket:websocket-servlet:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: javax.servlet:javax.servlet-api:3.1.0" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.eclipse.jetty:jetty-http:9.3.10.v20160621" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: de.flapdoodle.embed:de.flapdoodle.embed.mongo:1.50.5" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: de.flapdoodle.embed:de.flapdoodle.embed.process:1.50.2" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: commons-io:commons-io:2.4" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.apache.commons:commons-lang3:3.1" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: net.java.dev.jna:jna:4.0.0" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: net.java.dev.jna:jna-platform:4.0.0" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.apache.commons:commons-compress:1.10" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.springframework:spring-test:4.3.6.RELEASE" level="project" />
    <orderEntry type="library" name="Maven: org.codehaus.groovy:groovy-all:2.4.7" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: junit:junit:4.12" level="project" />
    <orderEntry type="library" scope="TEST" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
  </component>
</module>