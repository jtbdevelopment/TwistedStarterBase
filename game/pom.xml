<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <parent>
        <artifactId>TwistedStarterBase</artifactId>
        <groupId>com.jtbdevelopment.TwistedStarterBase</groupId>
        <version>1.0-SNAPSHOT</version>
    </parent>
    <modelVersion>4.0.0</modelVersion>

    <artifactId>game</artifactId>

    <dependencies>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-mongo</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-hazelcast</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-web</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-web-std-multiplayer</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-common</groupId>
            <artifactId>core-spring-web-mongo</artifactId>
            <version>${core.common.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-websocket</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <!--  TODO - TSB are these what this game wants  -->
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-std-tracker</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-std-maintenance</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <!--  TODO - TSB are these what this game wants  -->
        <!--
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-events</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-push-notifications</artifactId>
            <version>${core.games.version}</version>
        </dependency>
        -->


        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-web</artifactId>
            <version>${core.games.version}</version>
            <scope>test</scope>
            <type>test-jar</type>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-mongo</artifactId>
            <version>${core.games.version}</version>
            <type>test-jar</type>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.jtbdevelopment.core-games</groupId>
            <artifactId>games-dev-utilities</artifactId>
            <version>${core.games.devutilities.version}</version>
            <scope>test</scope>
        </dependency>
    </dependencies>

</project>