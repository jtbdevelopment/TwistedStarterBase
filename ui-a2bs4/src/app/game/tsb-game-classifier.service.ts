import {Injectable} from '@angular/core';
import {
    AbstractTurnClassifier,
    GameClassifier,
    MultiPlayerGame,
    PlayerService,
    StandardPhases
} from 'jtb-core-games-ui';

@Injectable()
export class TSBGameClassifier extends AbstractTurnClassifier implements GameClassifier<MultiPlayerGame> {

    private md5: string = '';

    constructor(private playerService: PlayerService) {
        super();
        this.playerService.player.subscribe(p => {
            this.md5 = (p ? p.md5 : '') || '';
        });
    }

    public classifyGame(game: MultiPlayerGame): string {
        //  TODO - TSB - if this game has a setup phase
        //  TODO - TSB - other nuances in phases
        let action = game.gamePhase === StandardPhases.Playing || game.gamePhase === StandardPhases.RoundOver;
        if (game.gamePhase === StandardPhases.Challenged && game.playerStates[this.md5]) {
            action = game.playerStates[this.md5] === 'Pending';
        }
        if (action) {
            return AbstractTurnClassifier.YOUR_TURN;
        }

        if (game.gamePhase === StandardPhases.Declined || game.gamePhase === StandardPhases.Quit || game.gamePhase === StandardPhases.NextRoundStarted) {
            return AbstractTurnClassifier.OLDER_GAMES;
        }

        return AbstractTurnClassifier.THEIR_TURN;
    }

}

