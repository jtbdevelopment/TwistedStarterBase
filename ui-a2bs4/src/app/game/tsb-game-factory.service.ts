import {Injectable} from '@angular/core';
import {GameFactory} from '../core-games-ui/games/gamefactory.serviceinterface';
import {TSBGame} from './tsb-game.model';

@Injectable()
export class TSBGameFactory implements GameFactory {
    public newGame(original?: any): any {
        return new TSBGame(original);
    }
}
