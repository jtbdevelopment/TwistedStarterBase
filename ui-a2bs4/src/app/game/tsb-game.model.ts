//  TODO - TSB - rename and add specific details

import {MultiPlayerGame} from 'jtb-core-games-ui';

export class TSBGame extends MultiPlayerGame {
  constructor(original?: any) {
    super(original);
    Object.assign(this, original);
  }
}
