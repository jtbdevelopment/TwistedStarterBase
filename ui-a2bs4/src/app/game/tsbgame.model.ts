import {MultiPlayerGame} from '../core-ui/games/multiplayergame.model';

//  TODO - TSB - rename and add specific details
export class TSBGame extends MultiPlayerGame {
    constructor(original?: any) {
        super(original);
        Object.assign(this, original);
    }
}
