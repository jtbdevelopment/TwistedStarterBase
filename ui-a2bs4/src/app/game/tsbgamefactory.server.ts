import {IGameFactory} from '../core-ui/games/igamefactory.service';
import {TSBGame} from './tsbgame.model';
import {Injectable} from '@angular/core';

@Injectable()
export class TSBGameFactory implements IGameFactory {
    public newGame(original?: any): any {
        return new TSBGame(original);
    }
}
