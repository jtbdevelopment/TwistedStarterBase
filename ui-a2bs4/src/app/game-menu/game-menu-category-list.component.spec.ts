import {TestBed, async, fakeAsync, tick} from '@angular/core/testing';
import {GameMenuCategoryListComponent} from './game-menu-category-list.component';
import {GameCacheService} from '../core-ui/gamecache/game-cache.service';
import {Observable, BehaviorSubject} from 'rxjs';
import {MultiPlayerGame} from '../core-ui/games/multi-player-game.model';
import {Game} from '../core-ui/games/game.model';
import {SinglePlayerGame} from '../core-ui/games/single-player-game.model';
import {Input, Component} from '@angular/core';
import {NgbModule, NgbCollapseModule} from '@ng-bootstrap/ng-bootstrap';

class MockGameCache {
    public static games: Map<string, BehaviorSubject<any[]>> = new Map<string, BehaviorSubject<any[]>>();

    public getGamesForCategory(category: string): Observable<any[]> {
        return MockGameCache.games.get(category);
    }
}

@Component({
    selector: 'game-menu-game-item',
    template: '<p>{{style}}</p><span>{{game.id}}</span>'
})
class MockGameItemComponent {
    @Input() public game: Game;
    @Input() public style: string;
}

describe('Component:  game menu category list component', () => {
    beforeEach(async(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbCollapseModule
            ],
            declarations: [
                GameMenuCategoryListComponent,
                MockGameItemComponent
            ],
            providers: [
                {provide: GameCacheService, useClass: MockGameCache}
            ]
        });
        TestBed.compileComponents();
    }));

    it('initializes and subscribes to category from game cache', fakeAsync(() => {
        const fixture = TestBed.createComponent(GameMenuCategoryListComponent);
        let expectedGames = [new Game({id: '1'}), new MultiPlayerGame({id: '2'}), new SinglePlayerGame({id: '3'})];
        let category = 'cat';
        let style = 's';
        MockGameCache.games.set(category, new BehaviorSubject([]));
        fixture.componentInstance.category = category;
        fixture.componentInstance.style = style;
        fixture.detectChanges();
        expect(fixture.componentInstance.games).toEqual([]);
        expect(fixture.componentInstance.isCollapsed).toBeFalsy();
        MockGameCache.games.get(category).next(expectedGames);
        tick();
        expect(fixture.componentInstance.games).toEqual(expectedGames);
        fixture.detectChanges();

        let querySelectorAll = fixture.nativeElement.querySelectorAll('game-menu-game-item');
        expect(querySelectorAll.length).toEqual(3);
        expect(querySelectorAll[0].textContent).toEqual(style + expectedGames[0].id);
        expect(querySelectorAll[1].textContent).toEqual(style + expectedGames[1].id);
        expect(querySelectorAll[2].textContent).toEqual(style + expectedGames[2].id);
    }));

    it('test toggle collapse', () => {
        const fixture = TestBed.createComponent(GameMenuCategoryListComponent);
        expect(fixture.componentInstance.isCollapsed).toBeFalsy();
        fixture.componentInstance.toggleCollapse();
        expect(fixture.componentInstance.isCollapsed).toBeTruthy();
        fixture.componentInstance.toggleCollapse();
        expect(fixture.componentInstance.isCollapsed).toBeFalsy();
    });
});
