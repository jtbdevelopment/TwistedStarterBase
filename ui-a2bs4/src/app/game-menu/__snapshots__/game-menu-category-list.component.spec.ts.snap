// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  game menu category list component initializes and subscribes to category from game cache 1`] = `
<app-game-menu-category-list
  category={[Function String]}
  gameCache={[Function MockGameCache]}
  games={[Function Array]}
  isCollapsed="false"
  style={[Function String]}
>
  <div
    class="category-wrapper"
  >
    
      
    <button
      class="list-group-item btn-group btn btn-group-s"
    >
      
          
      <span
        class="fa fa- float-left"
      />
      
          
      <span
        class="float-left"
      >
        cat
      </span>
      
          
      <span
        class="badge badge-secondary badge-pill align-middle float-right ml-auto"
      >
        3
      </span>
      
      
    </button>
    
      
    <div
      class="games-collapse collapse show"
      ng-reflect-collapsed="false"
    >
      
          
      
      <div
        class="btn-group-vertical btn-group-games"
      >
        
              
        <app-game-menu-game-item
          class="game-wrapper"
          ng-reflect-game="[object Object]"
          ng-reflect-style="s"
        >
          <p>
            s
          </p>
          <span>
            2
          </span>
        </app-game-menu-game-item>
        
          
      </div>
      <div
        class="btn-group-vertical btn-group-games"
      >
        
              
        <app-game-menu-game-item
          class="game-wrapper"
          ng-reflect-game="[object Object]"
          ng-reflect-style="s"
        >
          <p>
            s
          </p>
          <span>
            3
          </span>
        </app-game-menu-game-item>
        
          
      </div>
      <div
        class="btn-group-vertical btn-group-games"
      >
        
              
        <app-game-menu-game-item
          class="game-wrapper"
          ng-reflect-game="[object Object]"
          ng-reflect-style="s"
        >
          <p>
            s
          </p>
          <span>
            1
          </span>
        </app-game-menu-game-item>
        
          
      </div>
      
      
    </div>
    
  
  </div>
  
</app-game-menu-category-list>
`;
